larsim/EventGenerator/PhotonGen_module.cc

```diff
@@ -0,0 +1,250 @@
+    , fSigmaT{pset.get<double>("SigmaT")}
+    , fP{pset.get<double>("P")}
+    , fSigmaP{pset.get<double>("SigmaP")}
+    , fN{pset.get<int>("N")}
+    , fEngine(art::ServiceHandle<rndm::NuRandomService> {}->createEngine(*this, pset, "Seed"))
+  {
+    produces<sumdata::RunData, art::InRun>();
+    produces<std::vector<simb::MCTruth>>();
+
+    if (fFillTree) {
+      art::ServiceHandle<art::TFileService const> tfs;
+      fPhotonGen = tfs->make<TTree>("PhGen", "PhGen");
+      fPhotonGen->Branch("X", &(fShotPos[0]), "X/D");
+      fPhotonGen->Branch("Y", &(fShotPos[1]), "Y/D");
+      fPhotonGen->Branch("Z", &(fShotPos[2]), "Z/D");
+      fPhotonGen->Branch("T", &(fShotPos[3]), "T/D");
+      fPhotonGen->Branch("PX", &(fShotMom[0]), "PX/D");
+      fPhotonGen->Branch("PY", &(fShotMom[1]), "PY/D");
+      fPhotonGen->Branch("PZ", &(fShotMom[2]), "PZ/D");
+      fPhotonGen->Branch("PT", &(fShotMom[3]), "PT/D");
+      fPhotonGen->Branch("EventID", &fEvID, "EventID/I");
+    }
+
+    if (fScan) {
+      fPx = pset.get<double>("Px");
+      fPy = pset.get<double>("Py");
+      fPz = pset.get<double>("Pz");
+      std::cout << "Will generate photons from 3 points." << std::endl;
+    }
+  }
+
+  //____________________________________________________________________________
+  void PhotonGen::beginRun(art::Run& run)
+  {
+    art::ServiceHandle<geo::Geometry const> geo;
+    std::cout << "Number of optical detector: " << int(geo->Cryostat(0).NOpDet()) << std::endl;
+
+    //Boundary of Cryogenic
+    double CryoBounds[6];
+    geo->CryostatBoundaries(CryoBounds);
+    fXmin = CryoBounds[0];
+    fXmax = CryoBounds[1];
+    fYmin = CryoBounds[2];
+    fYmax = CryoBounds[3];
+    fZmin = CryoBounds[4];
+    fZmax = CryoBounds[5];
+    std::cout << "Cryo Boundaries:" << std::endl;
+    std::cout << "Xmin: " << fXmin << " Xmax: " << fXmax << " Ymin: " << fYmin << " Ymax: " << fYmax
+              << " Zmin: " << fZmin << " Zmax: " << fZmax << std::endl;
+    run.put(std::make_unique<sumdata::RunData>(geo->DetectorName()));
+
+    return;
+  }
+
+  //----------------------------------------------------------------
+  void PhotonGen::produce(art::Event& evt)
+  {
+    std::mt19937 rng;
+    rng.seed(std::random_device()());
+    std::uniform_real_distribution<double> distX(fXmin, fXmax);
+    std::uniform_real_distribution<double> distY(fYmin, fYmax);
+    std::uniform_real_distribution<double> distZ(fZmin, fZmax);
+    std::uniform_real_distribution<double> width(-2.0, 2.0);
+    srand((unsigned)time(NULL));
+
+    std::unique_ptr<std::vector<simb::MCTruth>> truthcol(new std::vector<simb::MCTruth>);
+    simb::MCTruth truth;
+    truth.SetOrigin(simb::kSingleParticle);
+
+    if (fScan) {
+      fX = distX(rng);
+      fY = fPy + width(rng);
+      fZ = fPz + width(rng);
+      Sample(truth);
+    }
+    else {
+      fX = distX(rng);
+      fY = distY(rng);
+      fZ = distZ(rng);
+
+      Sample(truth);
+    }
+
+    truthcol->push_back(truth);
```

Comment:
Suggest rewriting this as:
```cpp
fX = distX(rng);
if (fScan) {
  fY = fPy + width(rng);
  fZ = fPz + width(rng);
}
else {
  fY = distY(rng);
  fZ = distZ(rng);
}

truthcol->push_back(Sample());
```

where `Sample()` returns a `simb::MCTruth` object:

```cpp
simb::MCTruth Sample()
{
  simb::MCTruth mct;
  mct.SetOrigin(simb::kSingleParticle);
  // ... The code that you currently have in Sample()
  return mct;
}
```